### PR TITLE
Improve documentation for evaluate package

### PR DESCRIPTION
**Pull request by rjavadi**
# PR Type (Documentation)

## Short Description
Improving docs for ``evaluate`` module. 

I changed the main theme to provide easier navigation, and add same-page navigation on the right side navigation panel.
I'm going to add new experimental metrics in another PR.

### CODE REVIEW

**Comment by rjavadi** [comment]
@amrit110 I added new dependencies and generated a new ``poetry.lock``, what's the best way to resolve its conflicts?

**Comment by amrit110** [comment]
@rjavadi i've resolved the conflict now, had to merge against main and then just fix the `pyproject.toml` file manual and then do `poetry lock --no-update`. I've pushed the `sphinx-book-theme` dependency to be under the `docs` section. This will make sure that its not part of the cyclops package install but rather a development dependency.

**Comment by amrit110** [comment]
I feel the new theme improves navigation, but it breaks a few things because there were custom html files with footer and also the color scheme gets thrown ofcourse. The rest of the PR is good, @fcogidi should review more on how the user guide for the evaluate package reads.

**Review by fcogidi — COMMENTED** [review]
If it's possible, I'd suggest not having any `Evaluation` under the `User Guide` section, but as a separate section. Also, when the `Evaluation` section is clicked, it shows all the available modules and functions. I'd suggest not doing that, but instead, only expanding the drop-down and forcing the user to click on a specific module to see the details about it.

Inside the `User Guide` section, I'd put things like how to get started - installing cyclops, for example. For the evaluate package, things like how to implement a metric and how to set up the `evaluate` method would be useful in the `User Guide` section.

File: docs/source/conf.py [hunk 1]
```diff
@@ -84,7 +92,7 @@
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
 #
-html_theme = "theme"
+html_theme = "sphinx_book_theme"
 html_theme_path = ["."]
```
Comment (amrit110) [inline docs/source/conf.py:96]: I think this line can be removed and the `cyclops/docs/source/theme` directory can be removed once the new theme kicks in.

**Comment by rjavadi** [comment]
@fcogidi I'm going to make the changes. 
I still couldn't find a solution to get rid of that list of items that appear when clicking on ``Evaluation``.

**Review by amrit110 — APPROVED** [review]

**Pull request merged**